### DISCUSSION TITLE
Pull request merge commit details

### DISCUSSION THREAD

**Discussion by mikebowyer** [community/community#28801] [Pull Requests]
Hi,

I am having some issues understanding the best way to test pull request workflows in the context of the base(AKA target) branch. Ideally, a pull request workflow will merge the changes in the pull request branch with the latest branch tip of the base branch. 

However, in the GitHub action context of the pull request the merge commit which appears seems to be a result of a merge from the pull request branch tip, and the most recent common ancestor with the base branch. 

Is there some way to indicate that the merge which is desired would be from latest base branch commit, instead of the most recent common ancestor? 

Thanks in advance.

**Comment by airtower-luna** [comment]
> Is there some way to indicate that the merge which is desired would be from latest base branch commit, instead of the most recent common ancestor?

Not as such, the closest thing that comes to mind is setting up a branch protection rule with "Require branches to be up to date before merging". Then the PR will have to be rebased onto the current base branch (or have it merged into it) before the PR can be merged, guaranteeing that it's been tested with the current state of the base branch.

Technically you could also create the desired merge in your workflow, but that seems like a hassle. :sweat_smile:

**Comment by mikebowyer** [comment]
Thanks for the response @airtower-luna! 

The branch "Require branches to be up to date before merging" rule is not an ideal one, as this limits the maximum number of pull requests which can be merged in a day. For example, if pull request workflows take one hour to succeed, that means only 8 pull requests can be merged in an eight hour working day, which is not acceptable for larger projects. 

I was hoping to avoid performing the merge manually in our workflows, I expected this to be an option which GitHub or the [checkout action](https://github.com/actions/checkout/tree/v3) to provide for you.
